### PR TITLE
Feature/pvlan

### PR DESCRIPTION
**Pull request by davidefdl**
Change the trunk secondary and promisc to bool value. This will help implementing the provider side
with a simple true or false.

### CODE REVIEW

File: tests/test_interface_private_vlan.rb [hunk 1]
```diff
@@ -89,9 +89,8 @@ def test_interface_switchport_private_trunk_promisc
       end
       return
     end
-    interface.switchport_mode_private_vlan_trunk_promiscuous = ''
-    assert_equal(:promiscuous,
-                 interface.switchport_mode_private_vlan_trunk_promiscuous,
+    interface.switchport_mode_private_vlan_trunk_promiscuous = true
+    assert_equal(true, interface.switchport_mode_private_vlan_trunk_promiscuous,
```
Comment (chrisvanheuveln) [inline tests/test_interface_private_vlan.rb]:
```
-    assert_equal(true, interface.switchport_mode_private_vlan_trunk_promiscuous,
+    assert(interface.switchport_mode_private_vlan_trunk_promiscuous,
```

Change all below as well.

File: tests/test_interface_private_vlan.rb [hunk 2]
```diff
@@ -141,17 +139,15 @@ def test_interface_no_switchport_private_trunk_mode
...
-    assert_equal(:secondary,
-                 interface.switchport_mode_private_vlan_trunk_secondary,
+    interface.switchport_mode_private_vlan_trunk_secondary = true
+    assert_equal(true, interface.switchport_mode_private_vlan_trunk_secondary,
                  'Err: Switchport mode not as expected')
-    interface.switchport_mode_private_vlan_trunk_secondary = 'no'
-    assert_equal(:disabled,
-                 interface.switchport_mode_private_vlan_trunk_secondary,
+    interface.switchport_mode_private_vlan_trunk_secondary = false
+    assert_equal(false, interface.switchport_mode_private_vlan_trunk_secondary,
```
Comment (chrisvanheuveln) [inline tests/test_interface_private_vlan.rb]:
```
-    assert_equal(false, interface.switchport_mode_private_vlan_trunk_secondary,
+    refute(interface.switchport_mode_private_vlan_trunk_secondary,
```

**Comment by chrisvanheuveln** [comment]
Couple of nits but changes look good overall. Please get into the habit of stating in your PR description (including this one) which tests you've run and on which platforms.

File: tests/test_interface_private_vlan.rb (hunk 1 above)
Comment (davidefdl) [inline tests/test_interface_private_vlan.rb]: done

File: tests/test_interface_private_vlan.rb (hunk 2 above)
Comment (davidefdl) [inline tests/test_interface_private_vlan.rb]: done

File: tests/test_interface_private_vlan.rb [hunk 3]
```diff
@@ -273,10 +273,8 @@ def test_interface_switchport_pvlan_host_assoc_default
       return
     end
 
-    input = []
-    assert_equal(input,
-                 interface.switchport_mode_private_vlan_host_association,
-                 'Err: switchport private host_association not configured')
+    refute(interface.switchport_mode_private_vlan_host_association,
```
Comment (chrisvanheuveln) [inline tests/test_interface_private_vlan.rb]: whups - this should remain `assert_equal`, right?

**Comment by davidefdl** [comment]
Ran minitest towards 7k,9k,5k and 6k. 
Also for the default test case since feature private is not enabled, check for the right return values.

File: tests/test_interface_private_vlan.rb (hunk 3 above)
Comment (chrisvanheuveln) [inline tests/test_interface_private_vlan.rb]: I'm not sure you should have changed the assert_equal for these array tests. 

`refute` is testing for false, not empty array. You can use `refute_empty?` to fail if it's not an empty array. 

Be sure to retest on one of your test platforms after making these changes.

**Comment by davidefdl** [comment]
Use assert_nil for the default cases since feature pvlan is not enabled, the getter are returning nil.
Rested in platforms 7k, 9k, 5k and 6k

**Pull request merged**